src/components/ChannelSearch/SearchResults.tsx

```diff
@@ -1,18 +1,45 @@
 import React, { useCallback, useEffect, useState } from 'react';
 
+import { ChannelOrUserResponse, isChannelOrUserResponse } from './utils';
+
 import { Avatar } from '../Avatar/Avatar';
 import { useBreakpoint } from '../Message/hooks/useBreakpoint';
 
 import { useTranslationContext } from '../../context/TranslationContext';
 
-import type { UserResponse } from 'stream-chat';
-
 import type { DefaultUserType } from '../../types/types';
 
+export type DropdownContainerProps<Us extends DefaultUserType<Us> = DefaultUserType> = {
+  results: ChannelOrUserResponse[];
+  SearchResultItem: React.ComponentType<SearchResultItemProps<Us>>;
+  selectResult: (user: ChannelOrUserResponse) => Promise<void> | void;
+  focusedUser?: number;
+};
+
+const DefaultDropdownContainer = <Us extends DefaultUserType<Us> = DefaultUserType>(
+  props: DropdownContainerProps<Us>,
+) => {
+  const { focusedUser, results, SearchResultItem = DefaultSearchResultItem, selectResult } = props;
+
+  return (
+    <div>
+      {results.map((result: ChannelOrUserResponse, index: number) => (
+        <SearchResultItem
+          focusedUser={focusedUser}
+          index={index}
+          key={index}
+          result={result}
+          selectResult={selectResult}
+        />
+      ))}
+    </div>
+  );
+};
+
 export type SearchResultItemProps<Us extends DefaultUserType<Us> = DefaultUserType> = {
   index: number;
-  result: UserResponse<Us>;
-  selectResult: (user: UserResponse<Us>) => Promise<void> | void;
+  result: ChannelOrUserResponse;
```

Comment:
i'm assuming ChannelOrUserResponse needs all the generics for ChannelResponse/UserResponse